src/Native/Runtime/arm64/ExceptionHandling.S

```diff
@@ -1,4 +1,617 @@
+
+        ALLOC_CALL_FUNCLET_FRAME 0x60
+        stp d8, d9,   [sp, #0x00]
+        stp d10, d11, [sp, #0x10]
+        stp d12, d13, [sp, #0x20]
+        stp d14, d15, [sp, #0x30]
+        stp x0, x2,   [sp, #0x40]  // x0, x2 & x3 are saved so we have the exception object, REGDISPLAY and 
+        stp x3, xzr,  [sp, #0x50]  // ExInfo later, xzr makes space for the local "is_not_handling_thread_abort"
+
+#define rsp_offset_is_not_handling_thread_abort 0x58
+#define rsp_offset_x2 0x48
+#define rsp_offset_x3 0x50
+
+        //
+        // clear the DoNotTriggerGc flag, trashes x4-x6
+        //
+        INLINE_GETTHREAD x5 // x5 <- Thread*
+
+        ldr         x4, [x5, #OFFSETOF__Thread__m_threadAbortException]
+        sub         x4, x4, x0
+        str         x4, [sp, #rsp_offset_is_not_handling_thread_abort] // Non-zero if the exception is not ThreadAbortException
+
+        add         x12, x5, #OFFSETOF__Thread__m_ThreadStateFlags
+
+ClearRetry_Catch:
+        ldxr        w4, [x12]
+        bic         w4, w4, #TSF_DoNotTriggerGc
+        stxr        w6, w4, [x12]
+        cbz         w6, ClearSuccess_Catch
+        b           ClearRetry_Catch
+ClearSuccess_Catch:
+
+        //
+        // set preserved regs to the values expected by the funclet
+        //
+        RESTORE_PRESERVED_REGISTERS x2
+        //
+        // trash the values at the old homes to make sure nobody uses them
+        //
+        TRASH_PRESERVED_REGISTERS_STORAGE x2
+
+        //
+        // call the funclet
+        // 
+        // x0 still contains the exception object
+        blr         x1
+
+        EXPORT_POINTER_TO_ADDRESS PointerToRhpCallCatchFunclet2
+
+        // x0 contains resume IP
+
+        ldr         x2, [sp, #rsp_offset_x2]                    // x2 <- REGDISPLAY*
+
+// @TODO: add debug-only validation code for ExInfo pop
+
+        INLINE_GETTHREAD x1                                     // x1 <- Thread*
+
+        // We must unhijack the thread at this point because the section of stack where the hijack is applied
+        // may go dead.  If it does, then the next time we try to unhijack the thread, it will corrupt the stack.
+        INLINE_THREAD_UNHIJACK x1, x3, x12                      // Thread in x1, trashes x3 and x12
+
+        ldr         x3, [sp, #rsp_offset_x3]                    // x3 <- current ExInfo*
+        ldr         x2, [x2, #OFFSETOF__REGDISPLAY__SP]         // x2 <- resume SP value
+
+PopExInfoLoop:
+        ldr         x3, [x3, #OFFSETOF__ExInfo__m_pPrevExInfo]  // x3 <- next ExInfo
+        cbz         x3, DonePopping                             // if (pExInfo == null) { we're done }
+        cmp         x3, x2
+        blt         PopExInfoLoop                               // if (pExInfo < resume SP} { keep going }
+
+DonePopping:
+        str         x3, [x1, #OFFSETOF__Thread__m_pExInfoStackHead]     // store the new head on the Thread
+
+        adrp x3, RhpTrapThreads
```

Comment:
There is `PREPARE_EXTERNAL_VAR` macro that is supposed to encapsulate this. Should it be used here?

Also, the indentation is off.

(Same comment for the few other places in asm code that use `RhpTrapThreads`.)